app/services/listings/bump.rb

```diff
@@ -8,29 +8,20 @@ def initialize(listing, user:)
       @listing = listing
       @cost = listing.cost
       @user = user
-      @org = Organization.find_by(id: listing.organization_id)
     end
 
     def call
-      purchaser = [@org, @user].detect { |who| who&.enough_credits?(@cost) }
-      return false unless purchaser
+      @listing.purchase(@user) do |purchaser|
+        ActiveRecord::Base.transaction do
```

Comment:
Definitely like that this is in a transaction.  Is "enough_credits" the right response variable?  (Non-blocking, I was expecting a variable named "bought")

Reply:
If the `@listings.bump` call further down the transaction fails (it won't) they wouldn't be bought. But I see what you're saying and may rename it.